### PR TITLE
[Feature]Use multi hash tables to aggregate

### LINKED ISSUE #9428: [Enhancement]Use multi hash tables in aggregation node
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Partition of data to multi hash tables will reduce the size of each hash table and improve the performance.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes

Issue Number: close #9428

## Problem Summary:

SSB   Flat 100G
|query | 多哈希表 | 单哈希表|
|--|--|--|
|q1.1 | 94 | 121|
|q1.2 | 87 | 110|
|q.13 | 101 | 117|
|q2.1 | 431 | 535|
|q2.2 | 409 | 449|
|q2.3 | 299 | 331|
|q3.1 | 518 | 599|
|q3.2 | 357 | 406|
|q3.3 | 270 | 294|
|q3.4 | 52 | 62|
|q4.1 | 559 | 645|
|q4.2 | 212 | 241|
|q4.3 | 168 | 188|
|Total | 3557 | 4098|

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
Please reformat your BE C++ code:
http://doris.incubator.apache.org/zh-CN/developer-guide/cpp-format-code.html

**Comment by mrhhsg** [comment]
> Please reformat your BE C++ code: http://doris.incubator.apache.org/zh-CN/developer-guide/cpp-format-code.html

done.

**Comment by Gabriel39** [comment]
I still think we should have a detailed analysis on this.
I have some questions for SSB Flat 100G
1. fill factor and conflict factor in agg hash table (maybe an estimate value).
2. data characteristics. （such as data size and order）

I think there is a strong correlation between performance and workload. So we should take carefully to avoid potential performance fallback for other workloads.

File: be/src/vec/common/hash_table/hash_table_proxy_for_multi_tables.h [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    template <typename Derived, bool is_const>
+    class iterator_base {
+    public:
+        using it_type = std::conditional_t<is_const, typename Data::const_iterator,
+                                           typename Data::iterator>;
+
+    private:
+        using Container = std::conditional_t<is_const, const Self, Self>;
+
+        Container* container;
```
Comment (HappenLee) [inline be/src/vec/common/hash_table/hash_table_proxy_for_multi_tables.h:54]: Container* _container; Use underscores on member variables

**Pull request closed without merging**